canister/src/api/get_utxos.rs

```diff
@@ -85,14 +87,27 @@ pub fn get_utxos(request: GetUtxosRequest) -> Result<GetUtxosResponse, GetUtxosE
             (stats.ins_total / 10) as u128 * s.fees.get_utxos_cycles_per_ten_instructions,
             s.fees.get_utxos_maximum - s.fees.get_utxos_base,
         );
-        charge_cycles(fee);
+        if trusted {
+            charge_cycles(fee);
```

Comment:
@ielashi Because of this spot I am still in favor of using `trusted: bool` in this method instead of charging only in the trust version of the method since that will require a bit more refactoring since we will need to return one more value from this method. WDYT?